File: banking-component/src/test/java/de/rwth/swc/banking/TransferAggregator.java
package de.rwth.swc.banking;

import de.rwth.swc.interact.rest.RestMessage;
import org.junit.jupiter.api.extension.ParameterContext;
import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
import org.junit.jupiter.params.aggregator.ArgumentsAggregator;

import java.util.Map;
import java.util.Set;

public class TransferAggregator implements ArgumentsAggregator {

    public TransferAggregator() {
    }

    @Override
    public Object aggregateArguments(ArgumentsAccessor argumentsAccessor, ParameterContext parameterContext) throws ArgumentsAggregationException {
        return new RestMessage<>(
                Map.of(),
                Set.of(),
                new Transfer(
                        argumentsAccessor.getInteger(0),
                        argumentsAccessor.getString(1),
                        argumentsAccessor.getString(2)
                )
        );
    }
}

File: banking-component/src/test/kotlin/de/rwth/swc/banking/TransferAggregator.kt
package de.rwth.swc.banking

import de.rwth.swc.interact.rest.RestMessage
import org.junit.jupiter.api.extension.ParameterContext
import org.junit.jupiter.params.aggregator.ArgumentsAccessor
import org.junit.jupiter.params.aggregator.ArgumentsAggregationException
import org.junit.jupiter.params.aggregator.ArgumentsAggregator


class TransferAggregator : ArgumentsAggregator {
    @Throws(ArgumentsAggregationException::class)
    override fun aggregateArguments(
        argumentsAccessor: ArgumentsAccessor,
        parameterContext: ParameterContext
    ): RestMessage<Transfer> {
        return RestMessage(
            mutableMapOf(), mutableSetOf(),
            Transfer(
                argumentsAccessor.getString(0),
                argumentsAccessor.getString(1),
                argumentsAccessor.getInteger(2)
            )
        )
    }
}